Scaffolding: nullablility annotation issues

My VS build currently fails with this
```
4>D:\code\EntityFrameworkCore\src\EFCore.SqlServer\Scaffolding\Internal\SqlServerDatabaseModelFactory.cs(634,50,634,59): error CS8604: Possible null reference argument for parameter 'name' in 'string SqlServerDatabaseModelFactory.DisplayName(string? schema, string name)'.
4>D:\code\EntityFrameworkCore\src\EFCore.SqlServer\Scaffolding\Internal\SqlServerDatabaseModelFactory.cs(854,96,854,105): error CS8604: Possible null reference argument for parameter 'name' in 'string SqlServerDatabaseModelFactory.DisplayName(string? schema, string name)'.
4>D:\code\EntityFrameworkCore\src\EFCore.SqlServer\Scaffolding\Internal\SqlServerDatabaseModelFactory.cs(887,108,887,117): error CS8604: Possible null reference argument for parameter 'name' in 'string SqlServerDatabaseModelFactory.DisplayName(string? schema, string name)'.
4>D:\code\EntityFrameworkCore\src\EFCore.SqlServer\Scaffolding\Internal\SqlServerDatabaseModelFactory.cs(889,80,889,110): error CS8604: Possible null reference argument for parameter 'name' in 'DatabaseUniqueConstraint.DatabaseUniqueConstraint(DatabaseTable table, string name)'.
4>D:\code\EntityFrameworkCore\src\EFCore.SqlServer\Scaffolding\Internal\SqlServerDatabaseModelFactory.cs(925,86,925,95): error CS8604: Possible null reference argument for parameter 'name' in 'string SqlServerDatabaseModelFactory.DisplayName(string? schema, string name)'.
4>D:\code\EntityFrameworkCore\src\EFCore.SqlServer\Scaffolding\Internal\SqlServerDatabaseModelFactory.cs(927,58,927,77): error CS8604: Possible null reference argument for parameter 'name' in 'DatabaseIndex.DatabaseIndex(DatabaseTable table, string name)'.
4>D:\code\EntityFrameworkCore\src\EFCore.SqlServer\Scaffolding\Internal\SqlServerDatabaseModelFactory.cs(1005,59,1005,77): error CS8604: Possible null reference argument for parameter 'name' in 'string SqlServerDatabaseModelFactory.DisplayName(string? schema, string name)'.
4>D:\code\EntityFrameworkCore\src\EFCore.SqlServer\Scaffolding\Internal\SqlServerDatabaseModelFactory.cs(1020,63,1020,81): error CS8604: Possible null reference argument for parameter 'name' in 'string SqlServerDatabaseModelFactory.DisplayName(string? schema, string name)'.
4>D:\code\EntityFrameworkCore\src\EFCore.SqlServer\Scaffolding\Internal\SqlServerDatabaseModelFactory.cs(1025,68,1025,74): error CS8604: Possible null reference argument for parameter 'name' in 'DatabaseForeignKey.DatabaseForeignKey(DatabaseTable table, string name, DatabaseTable principalTable)'.
4>D:\code\EntityFrameworkCore\src\EFCore.SqlServer\Scaffolding\Internal\SqlServerDatabaseModelFactory.cs(1027,63,1027,77): error CS8604: Possible null reference argument for parameter 'onDeleteAction' in 'ReferentialAction? SqlServerDatabaseModelFactory.ConvertToReferentialAction(string onDeleteAction)'.
4>D:\code\EntityFrameworkCore\src\EFCore.SqlServer\Scaffolding\Internal\SqlServerDatabaseModelFactory.cs(1051,67,1051,85): error CS8604: Possible null reference argument for parameter 'name' in 'string SqlServerDatabaseModelFactory.DisplayName(string? schema, string name)'.

```
Source of confusion is this
https://github.com/aspnet/EntityFrameworkCore/blob/e2ba1c8dab28db719f7a58cacba98dabc23390a7/src/EFCore.SqlServer/Scaffolding/Internal/SqlDataReaderExtension.cs#L41-L48

The file is marked as nullable enable so my understanding is `T` is non-null but then it does not match annotation on method. On the other hand if this method can return null then it is being used to pass argument to a method which does not accept null.

--- a/src/EFCore.Relational/Scaffolding/Metadata/DatabasePrimaryKey.cs
+++ b/src/EFCore.Relational/Scaffolding/Metadata/DatabasePrimaryKey.cs
@@ -14,7 +14,7 @@ namespace Microsoft.EntityFrameworkCore.Scaffolding.Metadata
     /// </summary>
     public class DatabasePrimaryKey : Annotatable
     {
-        public DatabasePrimaryKey([NotNull] DatabaseTable table, [NotNull] string name)
+        public DatabasePrimaryKey([NotNull] DatabaseTable table, [CanBeNull] string? name)
         {
             Table = table;
             Name = name;
@@ -29,7 +29,7 @@ namespace Microsoft.EntityFrameworkCore.Scaffolding.Metadata
         /// <summary>
         ///     The name of the primary key.
         /// </summary>
-        public virtual string Name { get; [param: NotNull] set; }
+        public virtual string? Name { get; [param: CanBeNull] set; }
 
         /// <summary>
         ///     The ordered list of columns that make up the primary key.

--- a/src/EFCore.SqlServer/Scaffolding/Internal/SqlDataReaderExtension.cs
+++ b/src/EFCore.SqlServer/Scaffolding/Internal/SqlDataReaderExtension.cs
@@ -5,6 +5,8 @@ using System.Data.Common;
 using System.Diagnostics.CodeAnalysis;
 using JB = JetBrains.Annotations;
 
+#nullable enable
+
 namespace Microsoft.EntityFrameworkCore.SqlServer.Scaffolding.Internal
 {
     /// <summary>

--- a/src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs
+++ b/src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs
@@ -606,7 +606,7 @@ ORDER BY [table_schema], [table_name], [c].[column_id]";
             var tableColumnGroups = reader.Cast<DbDataRecord>()
                 .GroupBy(
                     ddr => (tableSchema: ddr.GetValueOrDefault<string>("table_schema"),
-                        tableName: ddr.GetValueOrDefault<string>("table_name")));
+                        tableName: ddr.GetFieldValue<string>("table_name")));
 
             foreach (var tableColumnGroup in tableColumnGroups)
             {
@@ -830,7 +830,7 @@ ORDER BY [table_schema], [table_name], [index_name], [ic].[key_ordinal]";
             var tableIndexGroups = reader.Cast<DbDataRecord>()
                 .GroupBy(
                     ddr => (tableSchema: ddr.GetValueOrDefault<string>("table_schema"),
-                        tableName: ddr.GetValueOrDefault<string>("table_name")));
+                        tableName: ddr.GetFieldValue<string>("table_name")));
 
             foreach (var tableIndexGroup in tableIndexGroups)
             {
@@ -976,7 +976,7 @@ ORDER BY [table_schema], [table_name], [f].[name], [fc].[constraint_column_id]";
             var tableForeignKeyGroups = reader.Cast<DbDataRecord>()
                 .GroupBy(
                     ddr => (tableSchema: ddr.GetValueOrDefault<string>("table_schema"),
-                        tableName: ddr.GetValueOrDefault<string>("table_name")));
+                        tableName: ddr.GetFieldValue<string>("table_name")));
 
             foreach (var tableForeignKeyGroup in tableForeignKeyGroups)
             {
@@ -989,7 +989,7 @@ ORDER BY [table_schema], [table_name], [f].[name], [fc].[constraint_column_id]";
                     .GroupBy(
                         c => (Name: c.GetValueOrDefault<string>("name"),
                             PrincipalTableSchema: c.GetValueOrDefault<string>("principal_table_schema"),
-                            PrincipalTableName: c.GetValueOrDefault<string>("principal_table_name"),
+                            PrincipalTableName: c.GetFieldValue<string>("principal_table_name"),
                             OnDeleteAction: c.GetValueOrDefault<string>("delete_referential_action_desc")));
 
                 foreach (var foreignKeyGroup in foreignKeyGroups)
@@ -1108,7 +1108,7 @@ WHERE name = '{connection.Database}';";
             return result != null ? Convert.ToByte(result) : (byte)0;
         }
 
-        private static string DisplayName(string? schema, string? name)
+        private static string DisplayName(string? schema, string name)
             => (!string.IsNullOrEmpty(schema) ? schema + "." : "") + name;
 
         private static ReferentialAction? ConvertToReferentialAction(string? onDeleteAction)

